doc/query.md

```diff
@@ -94,11 +150,60 @@ if (!name) {
 }
 ```
 
-`-ensureIndexed:withName:` returns the name of the index if it is successful,
-otherwise it returns `nil`.
+####Indexing for text search
+
+Since text search relies on SQLite FTS which is a compile time option, we must ensure that SQLite FTS is in fact available.  To verify that text search is enabled and that a text index can be created use `-isTextSearchEnabled` before attempting to create a text index.  If text search is not enabled see [compiling and enabling SQLite FTS][enableFTS] for details. 
+
+[enableFTS]: http://www.sqlite.org/fts3.html#section_2
+
+```objc
+if ([ds isTextSearchEnabled]) {
+    // Create a text index over the name and comment fields.
+    NSString *name = [ds ensureIndexed:@[@"name", @"comment"]
+                              withName:@"basic_text_index"
+                                  type:@"text"];
+    if (name == nil) {
+        // there was an error creating the index
+    }
+}
+```
+
+As text indexing relies on SQLite FTS functionality any custom tokenizers need to be managed through SQLite.  SQLite comes standard with the "simple" default tokenizer as well as a Porter stemming algorithm tokenizer ("porter").  Please refer to [SQLite FTS tokenizers][fts] for additional information on custom tokenizers.
+
+[fts]: http://www.sqlite.org/fts3.html#tokenizer  
+
+When creating a text index, overriding the default tokenizer setting is done by providing a `tokenize` parameter setting as part of the index settings.  The value should be the same as the tokenizer name given to SQLite when registering that tokenizer.  In the example below we set the tokenizer to `porter`.
+
+```objc
+if ([ds isTextSearchEnabled]) {
+    Map<String, String> settings = new HashMap<String, String>();
+    settings.add("tokenize", "porter");
+    // Create a text index over the name and comment fields.
+    // Setting the tokenizer to "porter".
+    NSString *name = [ds ensureIndexed:@[@"name", @"comment"]
+                              withName:@"basic_text_index"
+                                  type:@"text"
+                              settings:@{@"tokenize": @"porter"}];
+    if (name == null) {
+        // there was an error creating the index
+    }
+}
+```
+
+The `-ensureIndexed:...` methods returns the name of the index if 
+it is successful, otherwise they returns `nil`.
+
+##### Restrictions
+
+- There is a limit of one text index per datastore.
+- Text indexes cannot be created on field names containing an `=` 
+sign. This is due to restrictions imposed by SQLite's virtual table syntax.
 
-If an index needs to be changed, first delete the existing index, then call 
-`-ensureIndexed:withName:` with the new definition.
+####Changing and removing indexes
+
+If an index needs to be changed, first delete the existing index 
+by calling `-deleteIndexNamed:` and provide the index name of the index to be deleted, then call the appropriate 
+`-ensureIndexed:...` method with the new definition.
 
```

Comment:
So, I was thinking about when a dev wants to delete an index but doesn't have the name. They can call "listIndexes" and can inspect which fields were indexed and can then find the name of the index they want to delete. However, I had to look in the source code to figure this out (https://github.com/cloudant/CDTDatastore/blob/master/Classes/common/query/CDTQIndexManager.m#L148-L159). Could you include a little snippet here to show how to do that?